chcemvediet/apps/inforequests/tests/test_models/test_action.py

```diff
@@ -8,265 +8,267 @@
-                Action.TYPES.DISCLOSURE,
-                Action.TYPES.REFUSAL,
-                Action.TYPES.AFFIRMATION,
-                Action.TYPES.REVERSION,
-                Action.TYPES.REMANDMENT,
-                Action.TYPES.ADVANCED_REQUEST,
-                Action.TYPES.EXPIRATION,
-                Action.TYPES.APPEAL_EXPIRATION,
-                )
-        # Make sure we are testing all action types
-        defined_action_types = Action.TYPES._inverse.keys()
-        self.assertItemsEqual(tests, defined_action_types)
-
-        inforequest = self._create_inforequest()
-        branch = self._create_branch(inforequest=inforequest)
-        for action_type in tests:
-            action = self._create_action(branch=branch, type=action_type, deadline=3)
-            self.assertEqual(action.deadline, 3)
-
-    def test_deadline_field_with_default_value_if_omitted(self):
-        tests = (
-                (Action.TYPES.REQUEST,                8, dict()),
-                (Action.TYPES.CLARIFICATION_RESPONSE, 8, dict()),
-                (Action.TYPES.APPEAL,                30, dict()),
-                (Action.TYPES.CONFIRMATION,           8, dict()),
-                (Action.TYPES.EXTENSION,             10, dict()),
-                (Action.TYPES.ADVANCEMENT,         None, dict()),
-                (Action.TYPES.CLARIFICATION_REQUEST,  7, dict()),
-                (Action.TYPES.DISCLOSURE,            15, dict(disclosure_level=Action.DISCLOSURE_LEVELS.NONE)),
-                (Action.TYPES.DISCLOSURE,            15, dict(disclosure_level=Action.DISCLOSURE_LEVELS.PARTIAL)),
-                (Action.TYPES.DISCLOSURE,          None, dict(disclosure_level=Action.DISCLOSURE_LEVELS.FULL)),
-                (Action.TYPES.REFUSAL,               15, dict()),
-                (Action.TYPES.AFFIRMATION,         None, dict()),
-                (Action.TYPES.REVERSION,           None, dict()),
-                (Action.TYPES.REMANDMENT,            13, dict()),
-                (Action.TYPES.ADVANCED_REQUEST,      13, dict()),
-                (Action.TYPES.EXPIRATION,          None, dict()),
-                (Action.TYPES.APPEAL_EXPIRATION,   None, dict()),
+                (Action.TYPES.REQUEST,                Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, delivered_date, 8, Deadline.UNITS.WORKDAYS, None), dict()),
+                (Action.TYPES.CLARIFICATION_RESPONSE, Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, delivered_date, 8, Deadline.UNITS.WORKDAYS, None), dict()),
+                (Action.TYPES.APPEAL,                 Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, delivered_date, 15, Deadline.UNITS.CALENDAR_DAYS, None), dict()),
+                (Action.TYPES.CONFIRMATION,           Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, delivered_date, 8, Deadline.UNITS.WORKDAYS, None), dict()),
+                (Action.TYPES.EXTENSION,              Deadline(Deadline.TYPES.OBLIGEE_DEADLINE, delivered_date, 8+0, Deadline.UNITS.WORKDAYS, None), dict()),
```

Comment:
to `+0` je na zdoraznenie, ze akcia `EXTENSION` ma v sebe moznost takto predlzit deadline.

Reply:
@viliambalaz Viac by mi davalo zmysel, keby rucne vypocitas datum, kedy ktory deadline ma nastavat. Ten rucne vypocitany datum pouzijes v zozname ako `expected_deadline`. A porovnas, ze `action.deadline.deadline_date` je rovny tomu vypocitanemu datumu. Lebo teraz sa testuje vlastne iba to, ze skopirovany `Deadline` objekt je rovnaky ako v kode. Co nie je moc uzitocne.

Kludne na koniec ku kazdemu riadku v tabulke pridaj komentar tvaru: `# 10 WD sice delivered date` a podobne

Reply:
:ok: